geode/src/main/scala/akka/stream/alpakka/geode/internal/stage/GeodeQueryGraphLogic.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.geode.internal.stage
+
+import akka.stream.SourceShape
+import akka.stream.stage.StageLogging
+import org.apache.geode.cache.client.ClientCache
+import org.apache.geode.cache.query.SelectResults
+
+abstract class GeodeQueryGraphLogic[V](val shape: SourceShape[V], val clientCache: ClientCache, val query: String)
+    extends GeodeSourceStageLogic[V](shape, clientCache)
+    with StageLogging {
+
+  override def preStart(): Unit =
+    new Thread() {
```

Comment:
GraphStages that spawn threads should be avoided. Can you use `CqListenerAdapter` here as you did in `GeodeCQueryGraphLogic` to get notified on a new element?